### PR TITLE
Final updates for the thesis

### PR DESCRIPTION
**Pull request by mikicz**
1) Updated to Arca v0.2, once I publish the package, including setting timeouts
2) Add some missing tests
3) Fix some validation issues
4) Removed one extra condition

### CODE REVIEW

File: naucse/templates/_base_course.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
         {% if course is defined %}
             <header class="lesson-header">
                 <a href="{{ url_for('index') }}">Nauč se Python </a>
-                {% if course is defined and course.vars.get('coach-present') %}
+                {% if course.vars.get('coach-present') %}
```
Comment (hroncok) [inline naucse/templates/_base_course.html]: How is this relevant? Isn't it breaking anything?

File: naucse/models.py [hunk 2]
```diff
@@ -599,7 +599,8 @@ class CourseLink(CourseMixin, Model):
     branch: str = DataProperty(link, default="master")
 
     info = ForkProperty(repo, branch, entry_point="naucse.utils.forks:course_info",
-                        args=lambda instance: [instance.slug])
+                        args=lambda instance: [instance.slug],
+                        timeout=1)
```
Comment (hroncok) [inline naucse/models.py]: OK, can we rather keep it default and override it in travis config if it takes to long?

File: naucse/templates/_base_course.html (hunk 1 above)
Comment (mikicz) [reply naucse/templates/_base_course.html]: Well, it is not breaking anything, but the condition is redundant since the same one is three lines above that. But it was introduced by me in the main PR or in some follow-up...

File: naucse/models.py (hunk 2 above)
Comment (mikicz) [reply naucse/models.py]: This argument is passed along to the Task instance, meaning it cannot be overridden by environment variables

File: naucse/templates/_base_course.html (hunk 1 above)
Comment (hroncok) [reply naucse/templates/_base_course.html]: I see. OK. I'd rather see it in different commit but that's juts me having OCD.

File: naucse/models.py (hunk 2 above)
Comment (hroncok) [reply naucse/models.py]: I see. Let's keep it this way then. However, why 1 sec? Can it fail on us on Travis if it is a bit slower sometime?

Comment (mikicz) [reply naucse/models.py]: The timeout is only applied to task execution itself, meaning excluding startup of the container, I find it hard to believe that once the container is started this particular function would fail to complete in a second - it only loads a file, deserializes it from YAML, takes a couple of values and returns them in a JSON.

That said I am not dead set on having this timeout, I'm fine with removing it and keeping the 5-sec default from Arca...

Comment (hroncok) [reply naucse/models.py]: I'm not dead set against having it 1 second for now. I'm just saying: Be prepared to make it less strict once it starts being problematic. I've seen Travis CI builds where it got stuck for 2 seconds without any obvious reason. Cloud is somebody else's computer, you never know what is running there  during our deployment.

Comment (mikicz) [reply naucse/models.py]: Ok, we'll see what happens once I release v0.2 of Arca and the build in this branch won't fail because it can't install Arca...

File: naucse/templates/_base_course.html (hunk 1 above)
Comment (mikicz) [reply naucse/templates/_base_course.html]: Done

File: naucse/models.py (hunk 2 above)
Comment (hroncok) [reply naucse/models.py]: Would you like to use a tarball requirement for now before you do a release?

Comment (hroncok) [reply naucse/models.py]: Timeout in tests :)

Comment (mikicz) [reply naucse/models.py]: Was just about to comment on it. I will revert it to the default...

**Comment by mikicz** [comment]
The whole build was successful after the 1-second timeout was removed
